Use secret environment variable values in Kubernetes-based tasks (#2054)
astronomer/docs: astro/kubernetespodoperator.md

    - In the Amazon ECR Dashboard, click **Repositories** in the left menu.
    - Click the **Private** tab and then copy the URI of the repository that hosts the Docker image.

## Use secret environment variables with the KubernetesPodOperator

> Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running the KubernetesPodOperator, mount the secret to the Pod running the task.

Reviewer: Same as my comment for the k8s executor:

I think the nomenclature `<SECRET_KEY>` implies the use of an private key (e.g. the private key in a public/private key pair for API credentials). Could we instead say something like...

> Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running on the KubernetesExecutor, mount the secret to the Pod running the task. The name of an environment variable (`<ENV_VAR>` in the example below) can be used to create a Kubernetes secret in your DAG in order to access the environment variable's value at runtime.

> Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running the KubernetesPodOperator, mount the secret to the Pod running the task.

Author: ```suggestion
Astro [environment variables](environment-variables.md) marked as secrets are stored in a Kubernetes secret called `env-secrets`. To use a secret value in a task running on the KubernetesExecutor, you pull the value from `env-secrets` and mount it to the Pod running your task as a new Kubernetes Secret.
```

1. Add the following import to your DAG file:
   
    ```python
    from airflow.kubernetes.secret import Secret
    ```

> 2. Define a Kubernetes `Secret` in your DAG instantiation in the following format:

Reviewer 2: ```suggestion
2. Define a Kubernetes `Secret` in your DAG instantiation using the following format:
```

    ```python

>     secret_env = Secret(deploy_type="env", deploy_target="<SECRET_KEY>", secret="env-secrets", key="<SECRET_KEY>")

Author: ```suggestion
    secret_env = Secret(deploy_type="env", deploy_target="<VARIABLE_KEY>", secret="env-secrets", key="<VARIABLE_KEY>")
```

    namespace = conf.get("kubernetes", "NAMESPACE")
    ```
   

> 3. Reference the key for the environment variable, formatted as `$VARIABLE_KEY` in the task using the KubernetesPodOperator.

Reviewer 2: ```suggestion
3. Reference the key for the environment variable, formatted as `$VARIABLE_KEY` in the task using the `KubernetesPodOperator`.
```
Is the code formatting necessary here? I know sometimes with Kube/Docker stuff, the code formatting can vary.
